src/core/mac/sub_mac.cpp

```diff
@@ -396,6 +396,8 @@ Error SubMac::Send(void)
 {
     Error error = kErrorNone;
 
+    mTransmitFrame.SetTxPower(OT_RADIO_POWER_INVALID);
```

Comment:
I suggest setting this in `Mac::BeginTransmit()`, rather than here.

`BeginTransmit()` is where we initialize all other configurations on `txFrames` to their default values (e.g., channel, MaxCsmaBackoffs, MaxFrameRetries). Setting it there allows higher layers like `MeshForwarder` to modify these configurations while preparing the frame if needed.